empress/support_files/js/bp-tree.js

```diff
@@ -971,5 +971,71 @@ define(["ByteArray", "underscore"], function (ByteArray, _) {
         return this._nameToNodes[name];
     };
 
+    /**
+     * Returns a new BPTree object that contains just the tips (and ancestors)
+     * of the nodes in keepTips.
+     *
+     * This method was ported from iow.
+     * https://github.com/wasade/improved-octo-waddle/blob/master/bp/_bp.pyx#L732
```

Comment:
Assuming this was based on the current tip of the main repo:

```suggestion
     * https://github.com/wasade/improved-octo-waddle/blob/master/bp/_bp.pyx#L732
     * SHA: ac16703d2e00eb0abb560a647bff32be120311e1
```